From 54510127483ebe114b5a19d2f8d9f07abb6f3c1d Mon Sep 17 00:00:00 2001
From: Evan Huus <eapache@gmail.com>
Date: Sat, 25 Oct 2014 14:08:57 -0400
Subject: [PATCH 2/5] ncp2222: initialize buffer in build_expert_data

We do a bunch of conditional logic to fill in the buffer, and then were checking
strlen(buffer) to see if we'd actually filled it in or not, but if we hadn't
then the buffer was garbage and strlen(buffer) was throwing valgrind warnings.
Ensure this works as intended by setting the first byte of the buffer
unconditionally at the beginning, so strlen(buffer) returns 0 as expected in
that case.

Bug: 10628
Change-Id: I6f6c29fbbcb601b6ebccb3a4b312f0fa72c169ae
Reviewed-on: https://code.wireshark.org/review/4925
Reviewed-by: Evan Huus <eapache@gmail.com>
Reviewed-on: https://code.wireshark.org/review/5337
Reviewed-by: Michael Mann <mmann78@netscape.net>
---
 epan/dissectors/packet-ncp2222.inc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/epan/dissectors/packet-ncp2222.inc b/epan/dissectors/packet-ncp2222.inc
index c055d48..b74b3b0 100644
--- a/epan/dissectors/packet-ncp2222.inc
+++ b/epan/dissectors/packet-ncp2222.inc
@@ -2730,6 +2730,7 @@ build_expert_data(proto_tree *ncp_tree, const char *hf_name, char *buffer,
     char            temp_buffer[256]="\0";
     gboolean        in_struct=FALSE;
 
+    buffer[0] = '\0';
     tree_loc = ncp_tree->first_child;
     for (tree_pointer=tree_loc; tree_pointer!=NULL; tree_pointer=tree_pointer->next)
     {
-- 
2.1.3

